src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -984,6 +985,10 @@ trait Contexts { self: Analyzer =>
 
     def isNameInScope(name: Name) = lookupSymbol(name, _ => true).isSuccess
 
+    def isPlainClassInJavaUnit(sym: Symbol): Boolean =
```

Comment:
There are many uses of `findMember` other than the ones updated in this PR. This PR only searches in companions when type-checking `Ident`s or `Select`s while type-checking a Java source file. For example, given

```
public class A { public static class T }
public class B extends A
```

A reference `B.T` resolves to `A.T` when it appears inside a Java source file, but should (?) give an error when inside a Scala source file. So we need to check `context.unit.isJava`, not only `sym.isJavaDefined`.

There's no `compilationUnit` on symbols, only `sourceFile`, but that's says where a symbol is defined. The `context.unit.isJava` check is about where the symbol is referenced.